src/Plugin/Field/FieldFormatter/ErrandServicesFormatter.php

```diff
@@ -0,0 +1,142 @@
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\helfi_tpr\Entity\ChannelTypeCollection;
+use Drupal\helfi_tpr\Entity\Service;
+
+/**
+ * Field formatter to render errand service maps.
+ *
+ * @FieldFormatter(
+ *   id = "tpr_service_err_channel_list",
+ *   label = @Translation("TPR - Errand Service Channels List"),
+ *   field_types = {
+ *     "entity_reference"
+ *   }
+ * )
+ */
+
+class ErrandServicesFormatter extends EntityReferenceEntityFormatter {
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+        'sort_order' => [],
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * Gets the channel types.
+   *
+   * @return \Drupal\helfi_tpr\Entity\ChannelTypeCollection
+   *   The channel type collection.
+   */
+  private function getChannelTypes() : ChannelTypeCollection {
+    return ChannelTypeCollection::createFromArray($this->getSetting('sort_order'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $form = parent::settingsForm($form, $form_state);
+
+    $form['sort_order'] = [
+      '#type' => 'table',
+      '#caption' => $this->t('Order'),
+      '#header' => [
+        $this->t('ID'),
+        $this->t('Label'),
+        $this->t('Weight'),
+      ],
+      '#tableselect' => FALSE,
+      '#tabledrag' => [
+        [
+          'action' => 'order',
+          'relationship' => 'sibling',
+          'group' => 'group-weight',
+        ],
+      ],
+    ];
+
+    foreach ($this->getChannelTypes() as $item) {
+      $form['sort_order'][$item->id] = [
+        '#attributes' => ['class' => ['draggable']],
+        '#weight' => $item->weight,
+        'id' => ['#plain_text' => $item->id],
+        'label' => ['#plain_text' => $item->label()],
+        'weight' => [
+          '#type' => 'weight',
+          '#title' => $this->t('Weight for @title', ['@title' => $item->label()]),
+          '#title_display' => 'invisible',
+          '#default_value' => $item->weight,
+          '#attributes' => ['class' => ['group-weight']],
+        ],
+      ];
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) : array {
+
+    if (!$items->getEntity() instanceof Service) {
+      throw new \InvalidArgumentException('The field can only be used with tpr_errand_service entities.');
+    }
+
+
+//    $elements = parent::viewElements($items, $langcode);
+    $services = $items->getEntity();
+    $channelTypes = $this->getChannelTypes();
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+
+    $elements = [];
+    $errand_services = $items->referencedEntities();
+
+    foreach ($errand_services as $errand_service) {
+      foreach ($errand_service->getChannels() as $channel) {
+
+
+      if (in_array($channel->getType(), $elements)) {
```

Comment:
use the channel type in the key of $elements .. this if currently doesn't do anything, as $elements currently have multi value arrays in it